Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -27,7 +27,7 @@ namespace Microsoft.Identity.Web
     public class TokenAcquisition : ITokenAcquisition
     {
         private readonly AzureADOptions _azureAdOptions;
-        private readonly ConfidentialClientApplicationOptions _applicationOptions;
+        private readonly ConfidentialClientApplicationOptionsWithClientCertificate _applicationOptions;
```

Comment:
I think this brings confusion because of the class name. Here we believe that `_applicationOptions` has certificate, but this is not 100% sure. Would it be possible to bring the prop `public X509Certificate2 ClientCertificate { get; set; }` to `ConfidentialClientApplicationOptions` class?